### PR TITLE
Make sure the Netgame state is set to prevent a crash

### PR DESCRIPTION
**Pull request by JohnAFernandez**
If the client tries to respawn a player when the server is trying to end the game it will lead to a crash on the client side.  This sets the client's game to the correct state when an end game packet is received and adds extra checks before respawning.

I am hoping for a little more testing before this is merged.

### CODE REVIEW

File: code/network/multi_respawn.cpp
Comment (notimaginative) [inline code/network/multi_respawn.cpp:652]: This needs to be indented.

File: code/network/multimsgs.cpp
Comment (notimaginative) [inline code/network/multimsgs.cpp:4065]: Only the server should update `Netgame.game_state` so this is might cause issues since there is a small window in which it could be reset. I'm sure there are probably a few other places in the code where it get set like this, but it's best to try and avoid doing it.

I believe that `multi_warpout_all_players()` already sets a `Net_player` flag to indicate warpout so it might be safer to check that instead.

**Review by notimaginative — CHANGES_REQUESTED** [review]

File: code/network/multimsgs.cpp
Comment (JohnAFernandez) [reply code/network/multimsgs.cpp]: Ok, changed, but I do still want to do a testing session before enabling it to be merged.

**Review by notimaginative — APPROVED** [review]
Looks good to me. Approved pending your tests to verify it's working the way you want.

**Pull request merged**
